partner_tag_smart_assignation/models/res_partner_category_extension.py

```diff
@@ -0,0 +1,22 @@
+from odoo import fields, models, api
+
+
+class ResPartnerCategoryExtension(models.Model):
+    _inherit = "res.partner.category"
+
+    author_of_the_tag = fields.Many2one('res.users', string="Author")
+    department_that_uses_the_tag = fields.Many2one('hr.department', string="Department")
+    description_of_the_tag = fields.Text(string="Description")
+    valid_until = fields.Date(string="Valid until")
```

Comment:
No need to specify the string attribute if it's the same as the field name (`_` is automatically replaced by spaces in the UI)
```suggestion
    valid_until = fields.Date()
```